server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -654,6 +655,26 @@ public final Boolean readOptionalBoolean() throws IOException {
         return map;
     }
 
+    /**
+     * Reads a {@link List} of {@code V}-type and then converts them to a {@code Map} using keyMapper.
+     *
+     * @param valueReader The value reader
+     * @param keyMapper function to create a key from a value
+     * @return Never {@code null}.
+     */
+    public <K, V> Map<K, V> readMapFromList(final Writeable.Reader<V> valueReader, final Function<V, K> keyMapper) throws IOException {
+        final int size = readArraySize();
+        if (size == 0) {
+            return Collections.emptyMap();
```

Comment:
I guess it would be better to return `Map.of`. It's a bit better than `Collections.emptyMap` because methods like `Map.copyOf` are aware that it's immutable and don't do any allocations.

Reply:
Yeah, I think I copied this from one of the existing methods originally.

I personally prefer newer `Map.of()` api as well as it is much cleaner and shorter. I also totally agree about `Map.copyOf` point. I was originally thinking that `Collections.unmodifiableMap()` would not wrap `Collections.emptyMap()` but this was not a correct assumption.

Reply:
I think we have something that checks we do not use some prohibited apis. I wonder if it is possible to add `Collections.[empty|singleton][List|Set|Map]` there to avoid this in all places?